docs: use dedicated Sphinx role to reference GitHub issue

qmonnet opened the pull request:
Introduce and use a dedicated Sphinx role to easily reference GitHub issues.

Advantages:
- Easier and more consistent to reference GitHub issues
- Fewer risks of typos in the URL or between the numbers in the link URL and caption
- Output more consistent with one uniform way to express those references

Drawback: When reading the RST text file, opening the issue is one more step away.

[Note: This is strictly about formatting in the documentation and does not require reviews by all teams marked as code owners other than docs-structure, apologies for the multiple review requests.]

qmonnet committed:
docs: conf.py: add :gh-issue: role to reference GitHub issues

Use the extlink Sphinx extension to add a role for referencing GitHub
issues in a uniform fashion. Writing :gh-issue:`1234` will automatically
convert into "GitHub issue 1234", with a link to the relevant issue.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: harmonise references to GitHub issues by using dedicated role

Use the dedicated :gh-issue:`<number>` role to reference GitHub issues
in the documentation. This makes it easier and more consistent to
reference GitHub issues, with fewer risks of typo in the URL or between
the numbers in the link URL and caption, and makes the output more
consistent with one uniform way to express those references.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: conf.py: use a variable for the URL of the GitHub repository

Avoid having the base URL for Cilium's GitHub repository multiple times
in the configuration file for the Documentation. Use a variable instead.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

joestringer commented on Documentation/gettingstarted/k3s.rst:
```diff
@@ -16,8 +16,8 @@ workloads in unattended, resource-constrained, remote locations or inside IoT
-supported on amd64 architecture with `ARM support planned <https://github.com/cilium/cilium/issues/9898>`_
-for a future release.
+supported on amd64 architecture with ARM support planned for a future release
+(See :gh-issue:`9898`).
```
Not related to this PR, but we should be good to remove / update this given that we will ship arm64 containers in v1.10.x.

qmonnet replied:
Right, I haven't checked the status of the different GH issues referenced.
I can update this paragraph, do you know if there are specific steps for arm64, or if the whole comment should go?

joestringer replied:
Maybe given the k3s context here it'd be worth mentioning that amd64 and arm64 are supported. That should be enough to imply that arm32 is not supported in case anyone is thinking about that (although I *think* upstream might be thinking about removing such support).

qmonnet replied:
Ok thanks, I updated it in a new commit, for clarity.

joestringer reviewed (approved):
Looks good :+1:

qmonnet committed:
docs: k3s: remove reference to GitHub issue for arm64 support

ARM containers will be shipped with v1.10. Remove the outdated
reference, and update the note to mention that both amd64 and arm64 are
supported.

Suggested-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

joestringer reviewed (approved):
Normally I'd nominate all docs changes for backport to the most recent branch, but given that v1.10 is around the corner and this is mostly cosmetic (+ last patch only applies for v1.10), I do not intend to nominate this PR for backport.
